src/main/java/com/google/sps/utils/User.java

```diff
@@ -0,0 +1,102 @@
+package com.google.sps.utils;
+import com.google.appengine.api.datastore.Entity;
+import java.lang.IllegalArgumentException;
+import java.util.List;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+  
+/**
+ * Wrapper around a datastore Entity representing a user.
+ *
+ * This adds some type safety and relieves the need for type casting when
+ * handling user entities.
+ */
+public final class User {
+
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  // The "kind" of entity that represents a user.
+  // Use `new Entity(User.USER_ENTITY_KIND)` to construct user entities.
+  public final static String USER_ENTITY_KIND = "user";
+
+  private Entity entity;
+
+  // properties for the leaderboard
+  String userID;
+  String userName;
+  int score;
+
+  public User(Entity userEntity) {
+    if (userEntity.getKind() != USER_ENTITY_KIND) {
+      throw new IllegalArgumentException(
+          "Attempted to interpret Entity of kind " + userEntity.getKind()
+          + " as a User.");
+    }
+    entity = userEntity;
+
+    // set values for the leaderboard
+    userID = (String) entity.getProperty("userID");
+    userName = (String) entity.getProperty("userName");
+    score = ((Number) entity.getProperty("score")).intValue();
```

Comment:
These are already stored in the Entity. Do you need to store them outside as well? Is it intentional that the getters return these value, but the setters don't change these values? If so, they should probably have different names to avoid confusion.

Reply:
It was intentional to have them stored outside of the entity, because these properties are displayed on the leader board, so they need to be accessible from the javascript. I did change the setters to also set these values, though.

Reply:
We talked "in person" and decided to get rid of the extra local variables. We don't need them unless we use GSON to convert `User` to JSON. And it's simpler to not use GSON, because GSON is not well matched to our use case.